### PR TITLE
[SelectModal] New features added - back button and first row

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview: ##
See [Referenced Design](https://xd.adobe.com/view/bab9a62d-3f2a-45f8-be0f-5680c474d9c1-efea/screen/4f5c890c-a48d-4a55-8d5d-4d400dd9222e/specs/)
New features added - back button and first row:
* First row match the last part of the breadcrumb.
* First row is selectable
* Back button moves up.
* In search mode, back button go to root of the system.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2819](https://tacc-main.atlassian.net/browse/DES-2819)

## Summary of Changes: ##
* Updated file listing common component to take first row as optional parameter. This seems to be a good reliable way to handle the indent and go along with listing UX.
* Remove existing header component and add back button.
* Back button handler

## Testing Steps: ##
Use an app like https://designsafe.dev/rw/workspace/mpm?appVersion=1.1.0
1. Go to my data, navigate into folder and see the first row adjusted. Select the first row, ensure the text selection is right
2. Repeat the previous test case, use back button, observe the bread crumb and file listing updated.
3. Switch to community data (community data is public based and has different root, which makes it another test scenario). Repeat previous tests on community data.
4. Switch between My data and community and see if first row and back button states reset.

## UI Photos:

Highlighted portions show the relevant changes.

Screenshot 2024-07-10 at 9 38 22 AM

### CODE REVIEW

**Review by rstijerina — COMMENTED** [review]
One change request, can we disable selecting the root dir of the selected system? Might be good to try and prevent someone staging an entire system dir
Screenshot 2024-07-10 at 2 11 58 PM

**Review by rstijerina — APPROVED** [review]

**Comment by chandra-tacc** [comment]
> One change request, can we disable selecting the root dir of the selected system? Might be good to try and prevent someone staging an entire system dir.
Yes, will do that.

**Comment by chandra-tacc** [comment]
> > One change request, can we disable selecting the root dir of the selected system? Might be good to try and prevent someone staging an entire system dir.
> 
> Yes, will do that.

Addressed this and tested.
Below are screenshots of two scenarios:
Root selection is disabled:
Screenshot 2024-07-10 at 3 04 06 PM

Non-root is enabled:
Screenshot 2024-07-10 at 3 04 17 PM

File: client/modules/workspace/src/SelectModal/SelectModal.tsx [hunk 1]
```diff
@@ -109,83 +98,141 @@ const getScheme = (storageSystem: TTapisSystem): string => {
...
       align: 'end',
       title: '',
-      render: (_, record) => {
-        const shouldRenderSelectButton =
+      render: (_, record, index) => {
+        const selectionModeAllowed =
           (record.type === 'dir' && selectionMode === 'directory') ||
           (record.type === 'file' && selectionMode === 'file') ||
           selectionMode === 'both';
+        const isNotRoot =
```
Comment (rstijerina) [inline client/modules/workspace/src/SelectModal/SelectModal.tsx:233]: Can we still allow project system roots to be added?

Screenshot 2024-07-10 at 3 19 58 PM

File: client/modules/workspace/src/SelectModal/SelectModal.tsx [hunk 2]
```diff
@@ -109,83 +98,141 @@ const getScheme = (storageSystem: TTapisSystem): string => {
...
+            }
           >
-            <i
-              role="none"
-              className="fa fa-folder-o"
-              style={{ color: '#333333', marginRight: '8px' }}
-            ></i>
-            {data}
+            <LeftOutlined />
+            Back
```
Comment (rstijerina) [inline client/modules/workspace/src/SelectModal/SelectModal.tsx:177]: Should "Back" be disabled if you are currently in root?

Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:179]: Fixed.

File: client/modules/workspace/src/SelectModal/SelectModal.tsx (hunk 1 above)
Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:235]: Fixed this. And tested root selection in projects and ensured it is still not showing in other systems.

Screenshot 2024-07-11 at 8 07 29 AM

File: client/modules/workspace/src/SelectModal/SelectModal.tsx (hunk 2 above)
Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:179]: Screenshot 2024-07-11 at 8 18 32 AM
Screenshot 2024-07-11 at 8 18 26 AM

**Review by rstijerina — APPROVED** [review]

**Pull request merged**